### DISCUSSION TITLE
Control of the feature set for a MsiPackage

### DISCUSSION THREAD

**Discussion by harper23** [wixtoolset/issues#7623] [Questions]
I have to install two MSI packages. One MSI is for x86 and the other for
x64. I authored a bundle with a custom UI that contains a checkbox cb1.
With the following chain, cb1 controls the feature set of the x86 MSI to
be installed. If cb1 is not checked, the ProductFeature shall be
installed. If cb1 is checked, all features shall be installed.

    <Chain>
      <MsiPackage SourceFile="X86.msi" InstallCondition="cb1 = 0" >
        <MsiProperty Name="ADDLOCAL" Value="ProductFeature"/>
      </MsiPackage>
      <MsiPackage Id="SetupAll" SourceFile="X86.msi"
     InstallCondition="cb1 = 1" />
      <MsiPackage SourceFile="X64.msi" InstallCondition="VersionNT64" />
    </Chain>

This worked with WiX 3.x, but the migration to WiX 4 fails with the error:
  The 'ADDLOCAL' MsiProperty is controlled by the
  bootstrapper and cannot be authored.

How can I control the feature set for a MsiPackage in WiX 4?

**Comment by mvollmer1** [comment]
I use this same method to preset features as I start my MSI install from the boostrapper. 
Now that this is no longer supported, I'm wondering what the recommended alternative might be?

**Reply by robmen** [reply]
Same as in WiX v3, use the Bundle's MsiPackageFeature callbacks.

**Reply by mvollmer1** [reply]
Thank you.  I'll look into that.
